### ISSUE TITLE
Support Iceberg Nessie Catalog OAuth2 Authentication Type

### ISSUE THREAD

**Issue by nqvuong1998** [trinodb/trino#20838]
Currently, the iceberg.nessie-catalog.authentication.type exclusively supports the BEARER authentication type. Consequently, when configuring the Nessie catalog for Iceberg tables, it is essential to provide a token.
- In the case of token refresh, obtaining a new token and restarting the Trino cluster are necessary.
- Setting no expiration date is considered an anti-pattern in a production environment.
I am requesting the addition of a new authentication type for the Nessie catalog: OAuth2 (details available at https://projectnessie.org/tools/client_config/#authentication-type-oauth2).

**Comment by nqvuong1998** [comment]
cc @ajantha-bhat

**Comment by ajantha-bhat** [comment]
Thanks for creating the issue. 

Few users in Trino slack also showed interest. 
It is in the roadmaps. We will prioritize it.
  
cc: @adutra, @dimas-b

**Comment by dimas-b** [comment]
Nessie clients added authorization code and device code flows recently. It would be nice to support them in Trino.

**Comment by kojderszymon** [comment]
Any update on this issue?

**Comment by chenyair** [comment]
I would love to see this feature or allow the token to be changed using a refresh token without the restart of the environment

**Comment by ajantha-bhat** [comment]
Nessie recently supported REST catalog path and we are actively working with community in improving the Auth capabilities of the REST catalog. 

And for the old `NessieCatalog` path, we don't have a plan to support oAuth2 from Trino yet

cc: @snazy, @adutra

**Comment by dimas-b** [comment]
If people still use the old `NessieCatalog` integration with Trino, supporting OAuth2 there is probably only a matter of upgrading the Nessie client and exposing configuration options.

**Comment by ajantha-bhat** [comment]
> If people still use the old NessieCatalog integration with Trino, supporting OAuth2 there is probably only a matter of upgrading the Nessie client and exposing configuration options.

True. But Trino doesn't accept property bag and we need to expose each configuration individually. Plus PR reviews are not active. Effort may not be worth it if the REST path is planning to support the same.

**Comment by dimas-b** [comment]
AFAIK, OAuth2 capabilities of the current Iceberg REST client are limited (certainly weaker than the Nessie client's capabilities).

We can certain help and improve the Iceberg REST client for OAtuh2, but I imagine it will result in new config properties required on the client side in that case too :) .... which is normal, IMHO: security requires discernment, which naturally leads to have to configure something that is validated during the client-server interactions.

**Comment by shohamyamin** [comment]
Please support the Oauth2 for Nessie this is crucial for using Nessie with Trino 

Whether this is with the Nessie as rest catalog or directly with the regular Nessie

**Comment by darzanebor** [comment]
+10

**Comment by adutra** [comment]
> If people still use the old NessieCatalog integration with Trino, supporting OAuth2 there is probably only a matter of upgrading the Nessie client and exposing configuration options.

+1

> Whether this is with the Nessie as rest catalog or directly with the regular Nessie

If I understand correctly, accessing Nessie via the Iceberg REST api (as a REST catalog) is a valid option? In this case, the Dremio Auth Manager has all the features you need:

https://github.com/dremio/iceberg-auth-manager

The advantage of this approach is that it works for any REST catalog.

**Comment by snazy** [comment]
@adutra IIUC it would just work if the Auth Manager jar is bundled with Trino's Iceberg plugin, correct?

**Comment by adutra** [comment]
> [@adutra](https://github.com/adutra) IIUC it would just work if the Auth Manager jar is bundled with Trino's Iceberg plugin, correct?

Correct.
